Jellyfin.Server.Implementations/Item/BaseItemRepository.cs

```diff
@@ -255,6 +255,23 @@ public IReadOnlyList<BaseItemDto> GetItemList(InternalItemsQuery filter)
         return dbQuery.AsEnumerable().Where(e => e is not null).Select(w => DeserialiseBaseItem(w, filter.SkipDeserialization)).ToArray();
     }
 
+    /// <inheritdoc />
+    public IReadOnlyList<string> GetSeriesPresentationUniqueKeys(InternalItemsQuery filter)
+    {
+        ArgumentNullException.ThrowIfNull(filter);
+        PrepareFilterQuery(filter);
+
+        using var context = _dbProvider.CreateDbContext();
+        IQueryable<BaseItemEntity> dbQuery = PrepareItemQuery(context, filter);
+
+        dbQuery = TranslateQuery(dbQuery, context, filter);
+
+        dbQuery = ApplyGroupingFilter(dbQuery, filter);
+        dbQuery = ApplyQueryPaging(dbQuery, filter);
+
+        return dbQuery.Where(q => q.SeriesPresentationUniqueKey != null).Select(q => q.SeriesPresentationUniqueKey!).AsEnumerable().Distinct().ToArray();
```

Comment:
Won't `AsEnumerable().Distinct()` cause it to do the filtering client-side rather than in SQL? It shouldn't be necessary to even have AsEnumerable, I think.
```suggestion
        return dbQuery.Where(q => q.SeriesPresentationUniqueKey != null).Select(q => q.SeriesPresentationUniqueKey!).Distinct().ToArray();
```

Reply:
Doing the distinct call in the db breaks ordering (see ApplyGroupingFiltering)